Service Accelerator Template for PostgreSQL on Microsoft Azure

danizheleva opened the pull request:
This is the Service Accelerator Template completed for PostgreSQL on Microsoft Azure. 

Closes #21

finos-admin commented:
Thank you for your contribution and **Welcome to our Open Source Community!**

To make sure your pull request is successful, we need all our contributors to be identifiable, but we couldn't parse the GitHub details of the following people : Daniela-G Zheleva

Luckily, resolving the issue is straightforward and you can resolve it by following the instructions below.

1. Check your git client is configured with a user email `git config --list | grep email`
2. If the user email is missing, run the following command, substituting with your git commit email address `git config --global user.email email@example.com`
3. Make sure your git commit email is configured on GitHub by [Setting your Commit Email Address](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address)
4. Then, amend the authors in your commit history by using `git commit --amend` to change your last commit.

Alternatively, use the slightly more complex `git reset --soft` and `git rebase` to checkout your changes, rewrite the commit history locally and (force) push changes to the downstream branch.

If you have any issues with the steps above, please email [help@finos.org](mailto:help@finos.org) so we can help you resolve before reviewing and accepting your pull request.

Thanks once again for the contribution and understanding.

_cc_ **@finos-admin**

mcleo-d commented:
Hi @danizheleva - It looks like your `git` client hasn't been configured with the email address associated with [your GitHub profile](https://github.com/danizheleva) and this is causing the FINOS CLA Bot to trigger the warning above.

Can you follow the instructions here https://github.com/finos/cloud-service-certification/pull/60#issuecomment-687087785 and make the commit again?

Many thanks 😸 

James.

danizheleva commented:
Comments from group review:
- use of microsoft certificates for TLS 
- Many parts can be abstracted into lower level services (later)

tmewett commented:
One thing it would be good to document going forward is the scope for an Accelerator, i.e. how specific do we want them to be w.r.t the service/product in question? This ties into the the question of what extent we separate other services within a cloud provider, but also concerns the level of technical detail (e.g. in this document there is a few paragraphs on TLS, a very general technology) and background information (e.g. there is a section on the background and requirements for BYOK).

You'll have to forgive me, I'm not familiar with financial industry regulations. Context is good but if others think that some information is out of scope, it could be worth cutting to make the document more concise. (As I am working on an Accelerator too it would be good to know your opinions on this.)

mcleo-d commented:
Hi @tmewett

A great place to start is by reviewing the Kubernetes Service Accelerator Template that was created by @alfredtommy for GCP and accepted by the Cloud Service Certification community.

https://github.com/finos/cloud-service-certification/blob/master/gcp/gke/kubernetes.md

You might find there are recognisable commonalities between Azure and GCP that help answer your questions. Also, @alfredtommy is great should you have any questions.

James.

fleadsom commented on azure/postgresql/ServiceApprovalAcceleratorPostgreSQL.md:
```diff
@@ -0,0 +1,416 @@
+            With Azure AD authentication, you can manage database user identities and other Microsoft services in a central location, which simplifies permission management.            
+            <br><br>
+            when using Azure AD for authentication, the security features that build in AAD can help to protect the login process. The AAD token can be obtained in various
+            ways which could help to protect most of the password attach. 
```
Is there a typo here? "Protect most of the password attach"

abdullahgarcia replied:
+1

fleadsom commented on azure/postgresql/ServiceApprovalAcceleratorPostgreSQL.md:
```diff
@@ -0,0 +1,416 @@
+    <td class="tg-fymr">Authorization</td>
+    <td class="tg-f8tv">
+        <p>
+            It is recommended to only assign AAD groups roles within postgres, and not to create indivisual users within postgres itself.            
```
```suggestion
            It is recommended to only assign AAD groups roles within postgres, and not to create individual users within postgres itself.            
```

fleadsom commented on azure/postgresql/ServiceApprovalAcceleratorPostgreSQL.md:
```diff
@@ -0,0 +1,416 @@
+            It is recommended to only assign AAD groups roles within postgres, and not to create indivisual users within postgres itself.            
+            <br><br>
+            When setting up AAD integration for Azure postgres, an AAD account (either group or user) needs to be specified as the azure ad administorator for the postgres server.             
+            Only the AAD users that are specified can initially connect and add more AAD user/group into postgres and assign database roles.            
```
```suggestion
            Only the AAD users that are specified can initially connect, add more AAD users/groups into postgres and assign database roles.            
```

abdullahgarcia replied:
+1

fleadsom commented on azure/postgresql/ServiceApprovalAcceleratorPostgreSQL.md:
```diff
@@ -0,0 +1,416 @@
+            <br><br>
+            Custom database roles with least privileged access is always preferable, logically grouped permissions can be used to ease the burdon of administration
+            <br><br>                        
+            When a AAD group is added into the postgres server, any member of the group can login and assume the database role that is assigned to the group. It thus recommended to create and add 
```
```suggestion
            When a AAD group is added into the postgres server, any member of the group can login and assume the database role that is assigned to the group. It is therefore recommended to create and add 
```

fleadsom commented on azure/postgresql/ServiceApprovalAcceleratorPostgreSQL.md:
```diff
@@ -0,0 +1,416 @@
+            Custom database roles with least privileged access is always preferable, logically grouped permissions can be used to ease the burdon of administration
+            <br><br>                        
+            When a AAD group is added into the postgres server, any member of the group can login and assume the database role that is assigned to the group. It thus recommended to create and add 
+            AAD groups only to postgres, and manage all users and role assignments using AAD, and eleminate the needs to create individual users within postgres.            
```
```suggestion
            AAD groups only to postgres, and manage all users and role assignments using AAD. This eleminates the needs to create individual users within postgres.            
```

fleadsom commented on azure/postgresql/ServiceApprovalAcceleratorPostgreSQL.md:
```diff
@@ -0,0 +1,416 @@
+    <td class="tg-f8tv">
+        <p>
+            Only authorised Database Administrators group members should be allowed to create or update or delete any instance of Azure database for PostgreSQL.
+            Additionally, this should happen in accordance to the defined deployment process as well (e.g. via a build service, etc.).
```
```suggestion
            Additionally, this should happen in accordance with the defined deployment process (e.g. via a build service, etc.).
```

danizheleva commented:
Hi everyone, I have made the changes which came out of our PR review meeting and @fleadsom has updated the PR to fix some left over spelling mistakes. Please review again and please merge if approved.

njwilliams commented on azure/postgresql/ServiceApprovalAcceleratorPostgreSQL.md:
```diff
@@ -0,0 +1,417 @@
+        </p>
+        <h4>Connecting from Azure</h4>
+        <p>
+            To allow applications from Azure to connect to your Azure Database for PostgreSQL server, Azure connections must be enabled. For example, to host an Azure Web Apps
```
This is written strangely. It starts by saying that you need to enable azure connections to achieve (connecting from other apps in Azure). But the end of the paragraph is actually recommending to disable azure connections (and the subsequent point on data prevention also says best practice is to disable).  The consistency of what is being recommended isn't clear.

abdullahgarcia commented on azure/postgresql/ServiceApprovalAcceleratorPostgreSQL.md:
```diff
@@ -0,0 +1,417 @@
+            Microsoft Azure Active Directory (Azure AD) authentication is a mechanism of connecting to Azure Database for PostgreSQL using identities defined in Azure AD.
+            With Azure AD authentication, you can manage database user identities and other Microsoft services in a central location, which simplifies permission management.            
+            <br><br>
+            when using Azure AD for authentication, the security features that build in AAD can help to protect the login process. The AAD token can be obtained in various
```
"When"

abdullahgarcia reviewed (changes requested):
Hi Daniela, 

The document looks great overall!

I only have the following observations:

- The following should be used consistently across the document: "Azure" and "AD".

- MINOR: Not sure if we should be using the term "Bank", given that there could be other sort of institutions using this document. Perhaps, "institution" or "organisation", the last one is used in a few places in the document.

- MINOR: I think we should put Key Management into a subsection of its own rather than sharing with BYOK and HYOK. 

- There's a recommendation to use Azure Monitor’s diagnostic log service, which looks good. The only unanswered question to me is how long does the logs live/are retained within the service? And perhaps a link to its documentation?

Let me know if you need me to dive further into any of these points.

Thanks!

danizheleva commented:
> Hi @danizheleva,
> 
> The document looks great overall!
> 
> I only have the following observations:
> 
> * The following should be used consistently across the document: "Azure" and "AD".
> * MINOR: Not sure if we should be using the term "Bank", given that there could be other sort of institutions using this document. Perhaps, "institution" or "organisation", the last one is used in a few places in the document.
> * MINOR: I think we should put Key Management into a subsection of its own rather than sharing with BYOK and HYOK.
> * There's a recommendation to use Azure Monitor’s diagnostic log service, which looks good. The only unanswered question to me is how long does the logs live/are retained within the service? And perhaps a link to its documentation?
> 
> Let me know if you need me to dive further into any of these points.
> 
> Thanks!

Hey Abdullah, 

Thanks a lot for reviewing! 

1. AD refers to Active Directory rather than Azure - I have changed to consistently reference either Azure Active Directory or AAD 
2. Agreed, changed to use 'organization'
3. Agreed, I have put it in a separate section.
4. The retention period can be adjusted - I have added a note on that and also provided a link
